### ISSUE TITLE
Changesets ran into trouble committing your files (CI only)

### ISSUE THREAD

**Issue by thdk** [changesets/changesets#1277]
## Affected Packages

`@changesets/cli`
`@changesets/git`

## Problem

`npx changeset version` won't commit the files and doesn't give any details about the actual error.

```
🦋  error Changesets ran into trouble committing your files
...
🦋  info npm info x
🦋  info npm info y
🦋  warn x is not being published because version 0.3.0 is already published on npm
🦋  warn y is not being published because version 0.3.0 is already published on npm
🦋  warn No unpublished projects to publish
$ git status
HEAD detached at 1273ea3
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	deleted:    .changeset/silver-plums-study.md
	modified:   packages/assets/CHANGELOG.md
	modified:   packages/assets/package.json
	modified:   packages/ui/CHANGELOG.md
	modified:   packages/ui/package.json
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   .npmrc
```

## Proposed solution

- process should exit if commit fails so we don't publish anything that's not commit / pushed to the main branch
- the error message with output of stderr should be logged

I saw some git commands throw an error while others (tag, commit) simply return `false` when something went wrong.

**Comment by minenwerfer** [comment]
Ran into this same problem today.
In my case, I have some git hooks configured to lint files before commiting them, so being able to see why the commit failed is absolutely essential. I think piping the error to stderr shouldn't be that hard to implement.

**Comment by magicds** [comment]
I also encountered this issue. I think after providing the error message, it's great to consider the possible solution mentioned in #1234, which is about configuring the commit_message format

**Comment by tutods** [comment]
Any solution for this?

**Comment by PerryFinn** [comment]
Indeed, I am currently unable to determine the specific cause of the error.

**Comment by aicest** [comment]
Same error I encountered yesterday.
<img width="1058" alt="image" src="https://github.com/user-attachments/assets/9d4f44cf-4bea-4965-b185-3e8401ee1741" />


Debugging in CI...
<img width="1056" alt="image" src="https://github.com/user-attachments/assets/9edc90d2-a2ff-4fbd-8f8e-0f7c8e3fad27" />


The [solution](https://github.com/actions/checkout?tab=readme-ov-file#push-a-commit-using-the-built-in-token) for my case, hope it's helpful to you.
```yaml
      - run: git config user.name "github-actions[bot]"
      - run: git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
```
